lib/commons/dom/focus-disabled.js

```diff
@@ -14,6 +14,23 @@ function focusDisabled(el) {
     return true;
   }
 
+  let parentNode = vNode.parent;
+  while (parentNode) {
+    // if a form element is in a legend, that element will not be disabled even if the fieldset is
+    // https://developer.mozilla.org/en-US/docs/Web/HTML/Element/fieldset
+    if (parentNode.props.nodeName === 'legend') {
+      parentNode = null;
+      continue;
+    }
+    if (
+      parentNode.props.nodeName === 'fieldset' &&
+      parentNode.hasAttr('disabled')
+    ) {
+      return true;
```

Comment:
This will return true even if the fieldset is part of a different shadow tree. However form fields inside a shadow root do not apply `disabled` from a parent fieldset.

```html
<fieldset disabled>
  <legend><label>This is strange<input name="hello"></label></legend>

  <label>this should be disabled <input></label>

  <div id="shadow"></div>
</fieldset>

<script src="/axe.js"></script>
<script>
  const shadow = document.querySelector('#shadow');
  const shadowRoot = shadow.attachShadow({mode: 'open'});
  shadowRoot.innerHTML = '<label>Shadow input <input></label>';
</script>
```

VirtualNodes have a `shadowId` property on them, so if the two are equal then we can return `true`, otherwise it won't apply.